WebAppManifestClient.js

```diff
@@ -0,0 +1,82 @@
+/*!
+ * Copyright (c) 2020 Digital Bazaar, Inc. All rights reserved.
+ */
+import {httpClient} from '@digitalbazaar/http-client';
+import {getWebAppManifestIcon} from './manifest.js';
+
+const DEFAULT_HEADERS = {Accept: 'application/manifest+json, application/json'};
+
+export class WebAppManifestClient {
+  constructor({
+    defaultHeaders,
+    httpsAgent,
+    manifestProxyHost,
+    manifestProxyPath = '/manifest',
+    manifestProxy = false
+  } = {}) {
+    this.defaultHeaders = {...DEFAULT_HEADERS, ...defaultHeaders};
+    this.httpsAgent = httpsAgent;
+    this.manifestProxyHost = manifestProxyHost;
+    this.manifestProxyPath = manifestProxyPath;
+    this.manifestProxy = manifestProxy;
+  }
+
+  async getManifest({origin}) {
```

Comment:
Can we have separate functions that get the manifest from the origin directly and that get it from the proxy as well -- so we can call those if desired? Then this function can call those helpers internally.